### DISCUSSION TITLE
Protection from DoS attacks

### DISCUSSION THREAD

**Discussion by Vort** [markqvist/Reticulum#536] [Help & Questions] [answered]
Hello.

From _"Reticulum provides secure digital communication that cannot be subjected to outside control, manipulation or censorship"_, I make conclusion that censors will not like this project.
How _"Reticulum is Unstoppable..."_ can be ensured if censor decides to launch denial-of-service (DoS) attack on Reticulum?
For example, by flooding nodes, publicly available via TCP protocol, with thousands of service packets per second _(I did not figured out yet which exactly packet types can be used for attack - announcements, probably)_.

**Comment by markqvist** [comment, marked as answer]
Announces are the only currently known viable DoS attack vector, and RNS includes several automatic mitigations and flood controls for exactly that scenario. Interestingly, several attempted DoS attacks have occurred on the testnet, both before and after those mitigations were put in place, and apart from a bit of lag in some clients, everything continued to run as intended.

Since Reticulum doesn't have any fixed (or even globally known) routing tables or predetermined paths to specific destinations, you can always just route packets over alternative paths, and if one part of the network is experiencing congestion (either intentional or unintentional), the announce propagation and path determination will pick up on this pretty quickly and route things over a more efficient (or just available) path.
